src/pages/Main.tsx

```diff
@@ -77,118 +74,109 @@ export const Home: FC = () => {
-                  { id: selection.id, selected: selection.selected },
-                )
-                .then((result) => {
-                  if (result.isSuccess) {
-                    registerReplyBoxEmailsAdditionsTargetAction(
-                      client,
-                      context.data.ticket.id,
-                      linkedIssues,
-                    );
-                  }
-                });
-            }
-          },
-        ),
-      )
+    match(action.name)
+      .with("jiraReplyBoxNoteAdditions", () => {
+        ((action as { payload: ReplyBoxSelection[] }).payload ?? []).forEach((selection: ReplyBoxSelection) => {
+          const ticketId = action.subject;
+
+          if (context?.data?.ticket.id) {
+            client?.setState(
+              ticketReplyNotesSelectionStateKey(ticketId, selection.id),
+              { id: selection.id, selected: selection.selected },
+            ).then((result) => {
+              if (result.isSuccess && context?.data?.ticket.id) {
+                registerReplyBoxNotesAdditionsTargetAction(
+                  client,
+                  context.data.ticket.id,
+                  linkedIssues,
+                );
+              }
+            });
+          }
+        });
+      })
+      .with("jiraReplyBoxEmailAdditions", () => {
+        ((action as { payload: ReplyBoxSelection[] }).payload ?? []).forEach((selection: ReplyBoxSelection) => {
+          const ticketId = action.subject;
+
+          if (context?.data?.ticket.id) {
+            client?.setState(
+                ticketReplyEmailsSelectionStateKey(ticketId, selection.id),
+                { id: selection.id, selected: selection.selected },
+              )
+              .then((result) => {
+                if (result.isSuccess && context?.data?.ticket?.id) {
+                  registerReplyBoxEmailsAdditionsTargetAction(
+                    client,
+                    context.data.ticket.id,
+                    linkedIssues,
+                  );
+                }
+              });
+          }
+        })
+      })
       .with("jiraOnReplyBoxNote", () => {
         const ticketId = action.subject;
-        const note = action.payload.note;
+        const note = (action as { payload: ReplyBoxOnReplyNote }).payload.note;
 
         if (!ticketId || !note || !client) {
           return;
         }
 
-        if (ticketId !== context?.data.ticket.id) {
+        if (ticketId !== context?.data?.ticket.id) {
           return;
         }
 
         client.setBlocking(true);
-        client
-          .getState<{ id: string; selected: boolean }>(
-            `tickets/${ticketId}/notes/*`,
-          )
+        client.getState<{ id: string; selected: boolean }>(`tickets/${ticketId}/notes/*`)
           .then((r) => {
             const issueIds = r
               .filter(({ data }) => data?.selected)
               .map(({ data }) => data?.id as string);
```

Comment:
This doesn't make sense, optionally chain but typecast as string. This surely means the value can be a string or undefined.

Reply:
fixed